pkg/tsdb/testdata/scenarios.go

```diff
@@ -517,9 +521,33 @@ func getRandomWalk(query *tsdb.Query, tsdbQuery *tsdb.TsdbQuery) *tsdb.QueryResu
 
 	queryRes := tsdb.NewQueryResult()
 	queryRes.Series = append(queryRes.Series, series)
+	attachLabels(query, queryRes)
 	return queryRes
 }
 
+func attachLabels(query *tsdb.Query, queryRes *tsdb.QueryResult) {
+	labelText := query.Model.Get("labels").MustString("")
+	if labelText == "" {
+		return
+	}
+
+	re := regexp.MustCompile(`\b(\w+)(!?=~?)"([^"\n]*?)"`)
+	tags := make(map[string]string)
+
+	byt := []byte(labelText)
+	re.ReplaceAllFunc(byt, func(s []byte) []byte {
+		exp := string(s)
+		idx := strings.Index(exp, "=")
```

Comment:
strings.SplitN is a little clearer IMO than slice indexing.

Reply:
got rid of regex and using SplitN